rasa/core/policies/policy.py

```diff
@@ -354,49 +344,55 @@ def _metadata_filename(cls) -> Text:
         """
         pass
 
-    def persist(self, path: Union[Text, Path]) -> None:
-        """Persists the policy to storage.
-
-        Args:
-            path: Path to persist policy to.
-        """
-        # not all policies have a featurizer
-        if self.featurizer is not None:
-            self.featurizer.persist(path)
+    def persist(self) -> None:
+        """Persists the policy to storage."""
+        with self._model_storage.write_to(self._resource) as path:
+            # not all policies have a featurizer
+            if self.featurizer is not None:
+                self.featurizer.persist(path)
 
-        file = Path(path) / self._metadata_filename()
+            file = Path(path) / self._metadata_filename()
 
-        rasa.shared.utils.io.create_directory_for_file(file)
-        rasa.shared.utils.io.dump_obj_as_json_to_file(file, self._metadata())
+            rasa.shared.utils.io.create_directory_for_file(file)
+            rasa.shared.utils.io.dump_obj_as_json_to_file(file, self._metadata())
 
     @classmethod
-    def load(cls, path: Union[Text, Path], **kwargs: Any) -> "Policy":
-        """Loads a policy from path.
-
-        Args:
-            path: Path to load policy from.
-
-        Returns:
-            An instance of `Policy`.
-        """
-        metadata_file = Path(path) / cls._metadata_filename()
-
-        if metadata_file.is_file():
-            data = json.loads(rasa.shared.utils.io.read_file(metadata_file))
-
-            if (Path(path) / FEATURIZER_FILE).is_file():
-                featurizer = TrackerFeaturizer.load(path)
-                data["featurizer"] = featurizer
-
-            data.update(kwargs)
-
-            return cls(**data)
-
-        logger.info(
-            f"Couldn't load metadata for policy '{cls.__name__}'. "
-            f"File '{metadata_file}' doesn't exist."
-        )
-        return cls()
+    def load(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
```

Comment:
💡 When we instantiate the `GraphComponent` for inference we hand them a `resource` which they can use to retrieve their trained version from the `model_storage`